src/Marten/QuerySession.cs

```diff
@@ -119,13 +119,10 @@ private T load<T>(object id)
         private void assertCorrectIdType<T>(object id)
         {
             var mapping = Tenant.MappingFor(typeof(T));
-            if (id.GetType() != mapping.IdType)
+            if (id.GetType() != mapping.IdType && (id.GetType() != typeof(int) || mapping.IdType != typeof(long)))
             {
-                if (id.GetType() == typeof(int) && mapping.IdType == typeof(long))
-                    return;
-
                 throw new InvalidOperationException(
-                    $"The id type for {typeof(T).FullName} is {mapping.IdType.Name}, but got {id.GetType().Name}");
+                    "The id type for {typeof(T).FullName} is {mapping.IdType.Name}, but got {id.GetType().Name}");
```

Comment:
Why was the `$` removed, isn't it not required for string interpolation?

Reply:
You have good eyes! Nice catch. Fixed.